com.hybris.hyeclipse.editor/src/com/hybris/hyeclipse/editor/copyright/CopyrightUpdaterCleanUp.java

```diff
@@ -0,0 +1,108 @@
+package com.hybris.hyeclipse.editor.copyright;
+
+import org.eclipse.core.runtime.Assert;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.ui.cleanup.CleanUpContext;
+import org.eclipse.jdt.ui.cleanup.CleanUpOptions;
+import org.eclipse.jdt.ui.cleanup.CleanUpRequirements;
+import org.eclipse.jdt.ui.cleanup.ICleanUp;
+import org.eclipse.jdt.ui.cleanup.ICleanUpFix;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+
+import com.hybris.hyeclipse.editor.copyright.constant.CopyrightConstants;
+import com.hybris.hyeclipse.editor.copyright.fix.CopyrightFix;
+
+/**
+ * Implementation of {@link ICleanUp} performing copyright clean up
+ */
+public class CopyrightUpdaterCleanUp implements ICleanUp {
+
+	private static final String ADD_COPYRIGHTS_DESCRIPTION = "Add Copyrights";
+	private static final String OVERRIDE_COPYRIGHTS_DESCRIPTION = "Override existing copyrights";
+	private CleanUpOptions options;
+	private RefactoringStatus status;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public RefactoringStatus checkPostConditions(final IProgressMonitor monitor) throws CoreException {
+		try {
+			if (status == null || status.isOK()) {
+				return new RefactoringStatus();
+			} else {
+				return status;
+			}
+		} finally {
+			status = null;
```

Comment:
Very strange approach... why you don't set `status = null` before return? or assign the value you are returning now to some variable and then just before returning it you can assign `null` to status variable